log-manager/src/main/java/io/airlift/log/RollingFileHandler.java

```diff
@@ -1,89 +1,231 @@
+ * limitations under the License.
+ */
 package io.airlift.log;
 
-import ch.qos.logback.core.AsyncAppenderBase;
-import ch.qos.logback.core.ContextBase;
-import ch.qos.logback.core.encoder.EncoderBase;
-import ch.qos.logback.core.rolling.RollingFileAppender;
-import ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP;
-import ch.qos.logback.core.rolling.TimeBasedRollingPolicy;
-import ch.qos.logback.core.util.FileSize;
-import com.google.common.math.LongMath;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.ByteStreams;
+import com.google.common.io.MoreFiles;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
 import io.airlift.units.DataSize;
 
-import java.io.File;
-import java.util.logging.ErrorManager;
+import javax.annotation.concurrent.GuardedBy;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.logging.Handler;
 import java.util.logging.LogRecord;
+import java.util.zip.GZIPOutputStream;
 
 import static io.airlift.units.DataSize.Unit.MEGABYTE;
-import static java.lang.String.format;
+import static java.lang.Math.toIntExact;
 import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.StandardCopyOption.ATOMIC_MOVE;
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
 import static java.util.logging.ErrorManager.CLOSE_FAILURE;
+import static java.util.logging.ErrorManager.FLUSH_FAILURE;
 import static java.util.logging.ErrorManager.FORMAT_FAILURE;
+import static java.util.logging.ErrorManager.GENERIC_FAILURE;
+import static java.util.logging.ErrorManager.OPEN_FAILURE;
 import static java.util.logging.ErrorManager.WRITE_FAILURE;
 
 final class RollingFileHandler
         extends Handler
 {
-    private static final String TEMP_FILE_EXTENSION = ".tmp";
-    private static final String LOG_FILE_EXTENSION = ".log";
-    private static final FileSize BUFFER_SIZE_IN_BYTES = new FileSize(new DataSize(1, MEGABYTE).toBytes());
+    private static final int MAX_OPEN_NEW_LOG_ATTEMPTS = 100;
+
+    public enum CompressionType
+    {
+        NONE(Optional.empty()),
+        GZIP(Optional.of(".gz"));
```

Comment:
or maybe

NONE("")
GZIP(".gz")

Reply:
I actually like the optional better as it communicates that there may not be an extension.